pkg/prometheus/promcfg_test.go

```diff
@@ -5856,9 +5882,6 @@ alerting:
 				nil,
 				nil,
 			)
-			if err != nil {
```

Comment:
Maybe we want to refine this assertion like 
```
expectErr := expected == ""
if expectErr && err == nil &&  {
  t.Fatal("prometheus config generation should return an error")
}

if !expectErr && err != nil {
  t.Fatal("prometheus config generation should not return an error")
}
```
Alternatively, we could also add a `expectErr` field to the test cases and use it instead of relying on the value of `expected`

Reply:
@fpetkovski Thank you :) . I updated the test